packages/block-editor/src/components/rich-text/file-paste-handler.js

```diff
@@ -3,16 +3,18 @@
  */
 import { createBlobURL } from '@wordpress/blob';
 
-export function filePasteHandler( images, html ) {
-	if ( html || images.length === 0 ) {
+export function filePasteHandler( files, html ) {
+	// Only process file if no HTML is present.
+	// Note: a pasted file may have the URL as plain text.
+	if ( html || files.length === 0 ) {
 		return undefined;
 	}
-	let filesHTML = '';
-	images.forEach( ( image ) => {
-		const file = image.getAsFile ? image.getAsFile() : image;
-		// eslint-disable-next-line no-console
-		window.console.log( 'Received item:\n\n', file );
-		filesHTML += `<img src="${ createBlobURL( file ) }">`;
-	} );
-	return filesHTML;
+
+	// eslint-disable-next-line no-console
+	window.console.log( 'Received items:\n\n', files );
+
+	return files
+		.filter( ( { type } ) => /^image\/(?:jpe?g|png|gif)$/.test( type ) )
```

Comment:
I don't think this will for the native files, they are file URL that point to the local file system on the device.
Can you do the filter before we get to this point?

Reply:
Oh, I see. What's the type in that case? It would be good to keep this image logic together.

Reply:
In other words, what does `files` look like for native?

Reply:
I see this comes straight from the event: `const { pastedText, pastedHtml, files } = event.nativeEvent;`.